File: 35. ValidSudoku/ValidSudoku(1).go
// Runtime: 4 ms, faster than 70.55% of Go online submissions for Valid Sudoku.
// Memory Usage: 2.8 MB, less than 69.82% of Go online submissions for Valid Sudoku.
func isValidSudoku(board [][]byte) bool {
    seen := []byte{}

	//rows
    for i:=0; i<9; i++ {
        for j:=0; j<9; j++ {
            if(Contains(seen, board[i][j])){
                return false
            }
            if(board[i][j] != '.'){
				seen = append(seen, board[i][j])
            }
		}
		seen = []byte{}
	}
	//cols
    for i:=0; i<9; i++ {
        for j:=0; j<9; j++ {
            if(Contains(seen, board[j][i])){
                return false
            }
            if(board[j][i] != '.'){
                seen = append(seen, board[j][i])
            }
		}
		seen = []byte{}
	}
	//grids
    for i:=0; i<3; i++ {
        for j:=0; j<3; j++ {
            for ii:=i*3; ii<i*3+3; ii++ {
                for jj:=j*3; jj<j*3+3; jj++ {
                    if(Contains(seen, board[ii][jj])){
                        return false
                    }
                    if(board[ii][jj] != '.'){
                        seen = append(seen, board[ii][jj])
                    }
				}
			}
			seen = []byte{}
		}
	}

	return true;
}
func Contains(array []byte, e byte) bool{
	for _, arrayElement := range array{
        if (arrayElement == e){
			return true
        }
	}
	return false
}

File: 35. ValidSudoku/ValidSudoku(1).cs
// Runtime: 112 ms, faster than 74.11% of C# online submissions for Valid Sudoku.
// Memory Usage: 27.2 MB, less than 93.70% of C# online submissions for Valid Sudoku.
public class Solution {
    public bool IsValidSudoku(char[][] board) {
        HashSet<char> seen = new HashSet<char>();

        //rows
        for (int i=0; i<9; i++){
            for (int j=0; j<9; j++){
                if(seen.Contains(board[i][j]))
                    return false;
                if(board[i][j] != '.')
                    seen.Add(board[i][j]);
            }
            seen.Clear();
        }
        //cols
        for (int i=0; i<9; i++){
            for (int j=0; j<9; j++){
                if(seen.Contains(board[j][i]))
                    return false;
                if(board[j][i] != '.')
                    seen.Add(board[j][i]);
            }
            seen.Clear();
        }
        //grids
        for (int i=0; i<3; i++){
            for (int j=0; j<3; j++){
                for (int ii=i*3; ii<i*3+3; ii++){
                    for (int jj=j*3; jj<j*3+3; jj++){
                        if(seen.Contains(board[ii][jj]))
                            return false;
                        if(board[ii][jj] != '.')
                            seen.Add(board[ii][jj]);
                    }
                }
                seen.Clear();
            }
        }

        return true;
    }
}